src/components/UpcomingQuizCard/UpcomingQuizCard.test.tsx

```diff
@@ -0,0 +1,13 @@
+import { render, screen, fireEvent } from '@testing-library/react';
+import { describe, it, expect, vi } from 'vitest';
+
+import { UpcomingQuizCard } from './UpcomingQuizCard';
+
+describe('UpcomingQuizCard', () => {
+  it('calls onRegister when button is clicked', () => {
+    const handleRegister = vi.fn();
+    render(<UpcomingQuizCard title="Quiz" date="2023-08-12" onRegister={handleRegister} />);
+    fireEvent.click(screen.getByTestId('register-button'));
```

Comment:
The test clicks the register button using getByTestId('register-button'). Prefer queries that reflect user behavior and accessibility, such as getByRole('button', { name: /register now/i }). Using accessible queries ensures your UI has the correct accessible name.

**Sugerencia:**
```typescript
Replace with an accessible query: `fireEvent.click(screen.getByRole('button', { name: /register now/i }));` Or use userEvent from @testing-library/user-event for more realistic interactions.
```